tests/src/Interop/MarshalAPI/FunctionPointer/GetDelForFcnPtr_Negative_Catchable.cs

```diff
@@ -80,6 +80,70 @@ public static int RunGetDelForFcnPtrTest()
             Console.WriteLine(e);
         }
 
+        // Delegate -> FcnPtr -> Delegate
+        try
+        {
+            VoidDelegate del = (VoidDelegate)Marshal.GetDelegateForFunctionPointer(fcnptr, typeof(VoidDelegate));
+            if (del.Target != md.Target)
+            {
+                retVal = 0;
+                Console.WriteLine("Failure - the Target of the funcptr->delegate should be null since the original method is static.");
+                Console.WriteLine(del.Target);
+            }
+
+            if (del.Method != md.Method)
+            {
+                retVal = 0;
+                Console.WriteLine("Failure - The Method of the funcptr->delegate should be equal to the MethodInfo of the original method.");
+                Console.WriteLine(del.Method);
+            }
+
+            // Try to call it
+            del();
+
+            Console.WriteLine("Pass - got a delegate for the function pointer.");
+        }
+        catch (Exception e)
+        {
+            retVal = 0;
+            Console.WriteLine("Failure - received exception while converting funcptr to delegate.");
+            Console.WriteLine(e);
+        }
+
+        // Native FcnPtr -> Delegate
+        try
+        {
+            // Intentionally using the fcnptr but moved by 1 byte. We need a native pointer which is readable (since CLR assumes this is a function pointer which can be called
+            // and reads the first few instructions to check if it's something it knows about).
+            // But we need something which is NOT a reverse-pinvoke thunk so that CLR won't be able to match/find it.
+            // Note that it really doesn't matter where the address comes from as long as it's readable and not one of the thunks.
+            VoidDelegate del = (VoidDelegate)Marshal.GetDelegateForFunctionPointer(fcnptr + 1, typeof(VoidDelegate));
```

Comment:
Suggestion: Could you just use Marshal.CoTaskMemAlloc to allow a native memory instead of fcnptr + 1..

Reply:
Good - changed.